packages/addon/src/channel_selector.ts

```diff
@@ -35,40 +40,95 @@ function fillChannels(data: ChannelDetails[], selected: string | null, messageCl
 window.addEventListener("load", () => {
   const parent = window.parent;
-  const logo = document.getElementById("logo")!!
+  const logo = document.getElementById("logo")!;
 
   const mc = new MessageChannel();
-  const myPort = mc.port1
-  myPort.start()
-
-  parent.postMessage({ type: "SelectorMessageInitialize" }, "*", [mc.port2]);
-
-  function changeSize(expanded: boolean) {
-    document.body.setAttribute("data-expanded", "" + expanded);
-    myPort.postMessage({ type: "SelectorMessageResize", expanded })
-  }
-
-  myPort.addEventListener("message", (e: MessageEvent) => {
-    if (e.data.type == 'SelectorMessageChannels') {
-      const details = e.data as SelectorMessageChannels
-      // console.log(JSON.stringify("CHANNEL DETAILS: " + JSON.stringify(details)))
-      channels = details.channels
-      channelId = details.selected
-
-      const selectedColor = (channelId ? (channels.find(c => c.id == channelId)?.displayMetadata?.color) : null) ?? 'black'
-      logo.style.fill = selectedColor
+  const myPort = mc.port1;
+  myPort.start();
+  myPort.onmessage = ({data}) => {
+    console.debug("Received message: ", data);
+    switch(data.type){
+      case "iframeHandshake": {
+        collapse();
+        break;
+      }
+      case "iframeChannels": {
+        logo.removeEventListener("click", expand);
+        const {userChannels, selected} = data.payload as IframeChannelsPayload;
+        fillChannels(userChannels, selected, (channelStr) => {
+          myPort.postMessage({
+            type: "iframeChannelSelected",
+            payload: {
+              selected: channelStr || null
+            }
+          });
+          collapse();
+        });
+        const selectedChannel = userChannels.find((c) => c.id === selected);
+        logo.style.fill = selectedChannel?.displayMetadata?.color ?? "white";
+        logo.addEventListener("click", expand);
+        break;
+      }
     }
-  })
-
-  logo.addEventListener("click", () => {
-    fillChannels(channels, channelId, (id) => {
-      changeSize(false)
-      myPort.postMessage({ type: "SelectorMessageChoice", channelId: id })
-    })
-
-    // ask the parent container to increase the window size
-    changeSize(true)
-  })
+  };
+
+  parent.postMessage({
+    type: "iframeHello",
+    payload: {
+      initialCSS: {
+        width: `${8*4}px`,
+        height: `${8*5}px`,
+        right: "2px",
+        bottom: "2px",
+        zIndex: "1000",
+        "z-index": "1000",
+        position: "fixed"
+
+      }
+    }
+  }, "*", [mc.port2]);
+
+  const expand = () => {
+    document.body.setAttribute("data-expanded", "true");
+    myPort.postMessage({
+      type: "iframeRestyle",
+      payload: {
+        updatedCSS: {
+          width: `100%`,
+          height: `100%`,
+          top: 0,
+          left: 0,
+          zIndex: "1000",
+          "z-index": "1000",
+          position: "fixed"
+        }
+      }
+    });
+  }
 
+  const collapse = () => {
+    myPort.postMessage({
```

Comment:
Same comment as above - I would set up the message as an object and apply typing, then post it.